float64[] sonar_range # len: 360/step_size, 0->front
float64[] angle